@xen-orchestra/proxy/src/app/mixins/backups/_listPartitions.js

```diff
@@ -28,12 +28,19 @@ const parsePartxLine = createParser({
 
 // returns an empty array in case of a non-partitioned disk
 export const listPartitions = async devicePath => {
-  const parts = await fromCallback(execFile, 'partx', [
-    '--bytes',
-    '--output=NR,START,SIZE,NAME,UUID,TYPE',
-    '--pairs',
-    devicePath,
-  ])
+  let parts
+  try {
+    parts = await fromCallback(execFile, 'partx', [
+      '--bytes',
+      '--output=NR,START,SIZE,NAME,UUID,TYPE',
+      '--pairs',
+      devicePath,
+    ])
+  } catch (error) {
+    // Partx results with code 1 which throws an error since v2.33
+    console.debug('listPartitions', error)
+    return []
+  }
 
   return parts
     .split(/\r?\n/)
```

Comment:
Move these transforms in the `try` block and you won't need to declare `parts` as mutable.

Reply:
What if `parsePartxLine` throws an error?

Reply:
You're right, that makes sense, maybe using `.catch()` to returns an empty string would be a better solution?